pinot-segment-local/src/main/java/org/apache/pinot/segment/local/data/manager/TableDataManager.java

```diff
@@ -323,4 +323,12 @@ default void onConsumingToDropped(String segmentNameStr) {
    */
   default void onConsumingToOnline(String segmentNameStr) {
   }
+
+  /**
+   * Return list of segment names that need to be refreshed along with reason.
+   * @param tableConfig Table Config of the table
+   * @param schema Schema of the table
+   * @return List of {@link NeedRefreshResponse} with segment names and reason for refresh
+   */
+  List<NeedRefreshResponse> getSegmentsForRefresh(TableConfig tableConfig, Schema schema);
```

Comment:
Would a name like getSegmentsOutOfSync be better. We use both reload and refresh to bring the segment in sync with table config, so we can separate the action from current state.

Reply:
`getDivergedSegments` or `getStaleSegments` ?

Reply:
**getStaleSegments** sounds good to me . Thanks Rajat. So we need to change the name for other data structures too (like NeedRefreshInfo)?

Reply:
Yes. All of them have to be changed.

Reply:
This is done